File: 0053-maximum-subarray/0053-maximum-subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        ans, subArrEnding=nums[0], nums[0]
        for i in range(1,len(nums)):
            # Loop through possible subarrays as mark which is largest.
            subArrEnding=max(subArrEnding+nums[i],nums[i])
            # Keep track of largest value of subarrays.
            ans=max(subArrEnding,ans)
        return ans
            

File: 0053-maximum-subarray/0053-maximum-subarray.go
func maxSubArray(nums []int) int {
    ans, subArrEnding:=nums[0], nums[0]
    for i:=1; i<len(nums); i++{
        // Loop through possible subarrays as mark which is largest.
        if subArrEnding+nums[i]>nums[i] {
            subArrEnding=subArrEnding+nums[i]
        }   else {subArrEnding=nums[i]}
        // Keep track of largest value of subarrays.
        if subArrEnding>ans {ans=subArrEnding}
    }
    return ans
}